src/workerd/io/observer.h

```diff
@@ -62,6 +62,12 @@ class RequestObserver: public kj::Refcounted {
     return kj::mv(client);
   }
 
+  // Used to record when a worker has a dynamic dispatch binding (Called on the dispatching side)
+  virtual void setHasDispatchBinding() {};
+
+  // Used to record what worker dispatched the current worker (Called by the dispatch target).
+  virtual void setDispatcherId(kj::Maybe<kj::String> id) {};
```

Comment:
non-blocking nit: does it need to be `kj::String` or can it be `kj::StringPtr`?

Reply:
It did need to be a `kj::String`, the object that it's coming from does not outlive the metrics object and is dropped before the worker starts executing.